mindsdb/interfaces/knowledge_base/controller.py

```diff
@@ -788,26 +829,46 @@ def add(
                 return kb
             raise EntityExistsError("Knowledge base already exists", name)
 
-        if embedding_model is None:
-            # create default embedding model
-            model_name = self._get_default_embedding_model(project.name, params=params)
-            params['default_embedding_model'] = model_name
-        else:
-            # get embedding model from input
+        embedding_model_params = params.get('embedding_model', None)
+        reranking_model_params = params.get('reranking_model', None)
+
+        if embedding_model:
             model_name = embedding_model.parts[-1]
 
+        elif embedding_model_params:
+            # Get embedding model from params.
+            # This is called here to check validaity of the parameters.
+            get_embedding_model_from_params(
```

Comment:
this line constructs the model, but it is not assigned to any variable and isn't used
should it be like this?

Reply:
@ea-rus The only reason that this is being done is to validate the parameters that are passed to the model and to catch any issues at creation time rather than when using it via an `INSERT` or something.

Reply:
@ea-rus The only reason that this is being done is to validate the parameters that are passed to the model and to catch any issues at creation time rather than when using it via an `INSERT` or something.